hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,302 @@
+  private static final Logger LOG = LoggerFactory.getLogger(HiveViewOperations.class);
+
+  private final String fullName;
+  private final String database;
+  private final String viewName;
+  private final FileIO fileIO;
+  private final ClientPool<IMetaStoreClient, TException> metaClients;
+  private final long maxHiveTablePropertySize;
+
+  HiveViewOperations(
+      Configuration conf,
+      ClientPool<IMetaStoreClient, TException> metaClients,
+      FileIO fileIO,
+      String catalogName,
+      TableIdentifier viewIdentifier) {
+    String dbName = viewIdentifier.namespace().level(0);
+    this.metaClients = metaClients;
+    this.fileIO = fileIO;
+    this.fullName = BaseMetastoreCatalog.fullTableName(catalogName, viewIdentifier);
+    this.database = dbName;
+    this.viewName = viewIdentifier.name();
+    this.maxHiveTablePropertySize =
+        conf.getLong(HIVE_TABLE_PROPERTY_MAX_SIZE, HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public void doRefresh() {
+    String metadataLocation = null;
+    Table table;
+
+    try {
+      table = metaClients.run(client -> client.getTable(database, viewName));
+      HiveViewOperations.validateTableIsIcebergView(table, fullName);
+
+      metadataLocation =
+          table.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+
+    } catch (NoSuchObjectException | NoSuchIcebergViewException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s.%s", database, viewName);
+      }
+    } catch (TException e) {
+      String errMsg =
+          String.format("Failed to get view info from metastore %s.%s", database, viewName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newView = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    boolean updateHiveView = false;
+    BaseMetastoreTableOperations.CommitStatus commitStatus =
+        BaseMetastoreTableOperations.CommitStatus.FAILURE;
+
+    try {
+      Table view = null;
+
+      try {
+        view = metaClients.run(client -> client.getTable(database, viewName));
+      } catch (NoSuchObjectException nte) {
+        LOG.trace("View not found {}.{}", database, viewName, nte);
+      }
+
+      if (view != null) {
+        if (!view.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+          throw new AlreadyExistsException(
+              "Table with same name already exists: %s.%s", database, viewName);
+        }
+
+        // If we try to create the view but the metadata location is already set, then we had a
+        // concurrent commit
+        if (newView
```

Comment:
This check doesn't look correct.  If this is a `newView` there shouldn't be an existing view definition.  What if this is an existing hive view that does not have a location set?  It seems that this should fail if a view is found but you're attempting to create a new one.

Reply:
As far as I can tell, I think this is borrowed from the HiveTableOperations::doCommit() code.  There it makes a bit more sense as we get a lock, do a check, then fail if there's an existing table (before we write our table).  

It seems the change to abstract out common logic is in https://github.com/apache/iceberg/pull/9461/files

Reply:
I tried to follow the [table](https://github.com/apache/iceberg/blob/70b7aa534b2c79ccd7b6c0e0fd1be980772bb20a/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L189) behaviour . How concurrent commits will be different in case of View ? Am i missing something here?